src/hooks/useApplicationActions.ts

```diff
@@ -568,6 +597,187 @@ const useApplicationActions = (
+    },
+    unknown
+  >(
+    async ({
+      userName,
+      clientAddress,
+      contractAddress,
+      maxDeviation,
+      allowedSps,
+      disallowedSPs,
+      newAvailableResult,
+    }) => {
+      const signatures = await submitClientAllowedSpsAndMaxDeviation(
+        clientAddress,
+        contractAddress,
+        allowedSps,
+        disallowedSPs,
+        maxDeviation,
+      )
+
+      return await postChangeAllowedSPs(
+        initialApplication.ID,
+        userName,
+        owner,
+        repo,
+        activeAddress,
+        signatures,
+        newAvailableResult,
+        maxDeviation,
+      )
+    },
+    {
+      onSuccess: (data) => {
+        setApiCalling(false)
+        if (data != null) updateCache(data)
+      },
+      onError: () => {
+        setApiCalling(false)
+        setMessage(null)
+      },
+    },
+  )
+
+  const mutationChangeAllowedSPsApproval = useMutation<
+    Application | undefined,
+    unknown,
+    { activeRequest: StorageProvidersChangeRequest; userName: string },
+    unknown
+  >(
+    async ({ activeRequest, userName }) => {
+      setMessage(`Searching the pending transactions...`)
+
+      const clientAddress = getClientAddress()
+
+      const addedProviders = activeRequest?.Signers.find(
+        (x) => x['Add Allowed Storage Providers CID'],
+      )?.['Add Allowed Storage Providers CID']
+
+      const removedProviders = activeRequest?.Signers.find(
+        (x) => x['Remove Allowed Storage Providers CID'],
+      )?.['Remove Allowed Storage Providers CID']
+
+      const maxDeviation = activeRequest['Max Deviation']
+        ? activeRequest['Max Deviation'].split('%')[0]
+        : undefined
+
+      const proposalTxs = await getChangeSpsProposalTxs(
+        clientAddress,
+        maxDeviation,
+        addedProviders,
+        removedProviders,
+      )
+
+      if (!proposalTxs) {
+        throw new Error(
+          'Transactions not found. You may need to wait some time if the proposal was just sent.',
+        )
+      }
+
+      const signatures: {
+        maxDeviationCid?: string
+        allowedSpsCids?: { [key in string]: string[] }
+        removedSpsCids?: { [key in string]: string[] }
+      } = {}
+
+      const wait = async (ms: number): Promise<void> => {
+        await new Promise((resolve) => setTimeout(resolve, ms))
+      }
+
+      for (let index = 0; index < proposalTxs.length; index++) {
+        const proposalTx = proposalTxs[index]
+
+        setMessage(`Preparing the '${proposalTx.cidName}' transaction...`)
+
+        await wait(2000)
+        const messageCID = await sendApproval(proposalTx.tx)
+
+        if (messageCID == null) {
+          throw new Error(
+            `Error sending the '${proposalTx.cidName}' transaction. Please try again or contact support.`,
+          )
+        }
+
+        setMessage(
+          `Checking the '${proposalTx.cidName}' transaction, It may take several seconds, please wait...`,
```

Comment:
```suggestion
          `Checking the '${proposalTx.cidName}' transaction, It may take a few minutes, please wait... Do not close this window.`,
```